File: Decrypt.java
import java.io.*;
import java.util.*;

public class Decrypt {
    public static void run() {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter encrypted file path: ");
        String path = sc.nextLine();

        try {
            byte[] data = java.nio.file.Files.readAllBytes(new File(path).toPath());

            System.out.println("Enter 3 shares (x y):");
            int[][] shares = new int[3][2];
            for (int i = 0; i < 3; i++) {
                shares[i][0] = sc.nextInt();
                shares[i][1] = sc.nextInt();
            }

            int key = SSSSen.shamirCombine(shares);

            // XOR-дешифрование
            byte[] decrypted = new byte[data.length];
            for (int i = 0; i < data.length; i++) {
                decrypted[i] = (byte)(data[i] ^ key);
            }

            File out = new File("decrypted.txt");
            try (FileOutputStream fos = new FileOutputStream(out)) {
                fos.write(decrypted);
            }
            System.out.println("File decrypted: " + out.getAbsolutePath());

        } catch (IOException e) {
            System.out.println("File error: " + e.getMessage());
        }
    }
}

File: decrypt.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <string>
#include <cstdint>
#include "decrypt.h"
#include "ssssen.h"
using namespace std;

int decrypt(string inFile) {
    string encFile = "encrypted.txt";
    string outFile = "decrypted.txt";

    vector<pair<int,int>> shares;
    cout << "Enter 3 parts of key (x y):\n";
    for (int i = 0; i < 3; i++) {
        int x, y;
        cin >> x >> y;
        shares.push_back({x, y});
    }

    int key = shamir_combine(shares);
    cout << "Key: " << key << endl;

    ifstream fin(encFile, ios::binary);
    if (!fin) {
        cerr << "Error " << encFile << endl;
        return 1;
    }
    vector<char> data((istreambuf_iterator<char>(fin)), istreambuf_iterator<char>());
    fin.close();

    // XOR
    for (char &c : data) {
        c ^= key;
    }

    ofstream fout(outFile, ios::binary);
    fout.write(data.data(), data.size());
    fout.close();

    cout << "File saved: " << outFile << endl;
    return 0;
}

